tests/test_estimate_params.py

```diff
@@ -905,6 +905,159 @@ def test_incorrect_lens(self):
+        if not all(abs(m.parameters[key] - gt[key]) > 0.02 for key in keys):
+            self.fail("m.parameter shouldn't be too close to the original parameters")
+
+        # When tolerance is a list of many integers
+        with self.assertRaises(ValueError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = [1, 2, 3])
+
+        # When tolerance is a list of mixed values (integers and strings)
+        with self.assertRaises(TypeError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = [1, '2', '3'])
+        
+        # When tolerance is a tuple
+        with self.assertRaises(ValueError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = (1, 2, 3))
+        
+        # Passing Tolerance and Method together should work
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys=keys, method = 'TNC', tol = 1e-9)
+        track1 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        # Using different methods should result in different errors.
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = 1e-9)
+        track2 = m.calc_error(results.times, results.inputs, results.outputs)
+        self.assertNotAlmostEqual(track1, track2)
+
+        # Tests checking very small tolerance values
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, 
+                          bounds=bound, keys=keys, method = 'TNC', tol = 1e-14)
+        track1 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        # Now testing what occurs with even smaller tolerance
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, 
+                            bounds=bound, keys=keys, method='TNC', tol=1e-20)
+        track2 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, 
+                            bounds=bound, keys=keys, method='TNC', tol=1e-9)
```

Comment:
Is this part necessary? I don't see the error calculated for this one and compared to any of the others, so we might be able to skip this. If we keep it, a comment as to what it is testing would be beneficial

Reply:
Nope, this is not necessary. It was going to be used for another comparison that was already happening elsewhere so I had intended to remove it.

Reply:
@aqitya should we delete it then?

Reply:
Oops. Thought I had deleted it already. Sorry, will push another commit with that bit of code removed.